### PR TITLE
swig: fix cross compilation

### PR DESCRIPTION
**Pull request by eburimu**
###### Motivation for this change

Fixes https://github.com/NixOS/nixpkgs/issues/51171

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg build pkgsCross.aarch64-multiplatform.buildPackages.swig

File: pkgs/development/tools/misc/swig/3.x.nix [hunk 1]
```diff
@@ -11,7 +11,7 @@ stdenv.mkDerivation rec {
     sha256 = "1wyffskbkzj5zyhjnnpip80xzsjcr3p0q5486z3wdwabnysnhn8n";
   };
 
-  nativeBuildInputs = [ autoconf automake libtool bison ];
+  nativeBuildInputs = [ autoconf automake libtool bison pcre.dev ];
```
Comment (Mic92) [inline pkgs/development/tools/misc/swig/3.x.nix]: Since it is in `nativeBuildInputs` I wonder why this is the case?

Comment (eburimu) [reply pkgs/development/tools/misc/swig/3.x.nix]: pcre-config is missing when pcre is in buildInputs. See pcre drv.

Comment (eburimu) [reply pkgs/development/tools/misc/swig/3.x.nix]: wow, rebuild linux 500+ :(

Comment (Mic92) [reply pkgs/development/tools/misc/swig/3.x.nix]: We can push this to staging.

**Comment by Mic92** [comment]
Ah it was already staging. Nevermind. I merged it in 8e97e8009f88380ec4975fb162969757d5037bdc

**Comment by matthewbauer** [comment]
This doesn't look right... you should not use buildPackages pcre-config. Instead, something like this is recommended:

```
PCRE_CONFIG = "${pcre.dev}/bin/pcre-config";
```

**Comment by Mic92** [comment]
well. This does not work either when pcre-config is compiled for the wrong architecture.

**Comment by matthewbauer** [comment]
It should be a bash script in this case at least

**Comment by Mic92** [comment]
But the shebang is still incorrect.

**Comment by matthewbauer** [comment]
Yeah ideally it would use /usr/bin/env sh in this case. Although it is safe for now to just assume the build platform will always be the same in nix. build1 -> target will be a different drv from build2-> target even though they should in theory be the same.

**Comment by Mic92** [comment]
There is no `/usr/bin/env` in the nix sandbox. `/bin/sh` has its own set of problems, in the sandbox it is a stripped down bash, on systems without a sandbox it might be a dash.

**Comment by Mic92** [comment]
An more general solution for the problem could be to have a build support hook that puts the right binary/variant in the PATH. Which means that pcre-config in `$dev/bin` remains the original one, but when we add it as a buildInput it would add a host-variant to `$PATH`. llvm-config has pretty similar problem but is a complex executable that cannot be easily re-implemented.

**Pull request merged**